### PR TITLE
Fix public-only subnets

### PR DESCRIPTION
**Pull request by Nuru**
## what
- Fix configuration of public subnets when private subnets are not created

## why
- Bug fix

### CODE REVIEW

File: nat-instance.tf [hunk 1]
```diff
@@ -124,6 +124,8 @@ resource "aws_eip_association" "nat_instance" {
   allocation_id = local.nat_eip_allocations[count.index]
 }
 
+# If private IPv4 subnets and NAT Gateway are both enabled, create a
```
Comment (aknysh) [inline nat-instance.tf:127]:
```suggestion
# If private IPv4 subnets and NAT instance are both enabled, create a
```

File: nat-instance.tf [hunk 2]
```diff
@@ -124,6 +124,8 @@ resource "aws_eip_association" "nat_instance" {
...
+# default route from private subnet to NAT Gateway in each subnet
```
Comment (aknysh) [inline nat-instance.tf:128]:
```suggestion
# default route from private subnet to NAT instance in each subnet
```

**Review by aknysh — COMMENTED** [review]
a few nitpicks

**Review by Benbentwo — APPROVED** [review]

**Pull request merged**